test/unit/assorted/server_selection_logic_spec_utils.ts

```diff
@@ -0,0 +1,151 @@
+import { Document, EJSON } from 'bson';
```

Comment:
"logic_spec_utils" implies to me these won't be used in the forthcoming integration tests? Could we keep all this logic inside the `server_selection.spec.test.ts` test file. I think if the logic isn't common to anything else keeping it all in one file is worth keeping things associated. But if this is reusable, then maybe the name should be more generic? My end goal is just making discoverability as clear as possible.

Reply:
Some of these functions might be reused in the forthcoming work and when that happens, I'll move them.  But each of the three test runners for server selection have a different test format, so a lot of the code here will not be reused.